homeassistant/components/publibike/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""The PubliBike Public API integration."""
+
+import logging
+
+from pypublibike.location import Location
+from pypublibike.publibike import PubliBike
+from pypublibike.station import Station
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    BATTERY_LIMIT,
+    DOMAIN,
+    LATITUDE,
+    LONGITUDE,
+    STATION_ID,
+    UPDATE_INTERVAL,
+)
+
+PLATFORMS = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PubliBike integration from a config entry."""
+
+    publi_bike = PubliBike()
+
+    station_id = entry.data.get(STATION_ID)
+    if station_id:
+        all_stations = await hass.async_add_executor_job(publi_bike.getStations)
+        station = [s for s in all_stations if s.stationId == station_id][0]
+    else:
+        lat = (
+            entry.data[LATITUDE]
+            if LATITUDE in entry.data.keys()
+            else hass.config.latitude
+        )
+        lon = (
+            entry.data[LONGITUDE]
+            if LONGITUDE in entry.data.keys()
+            else hass.config.longitude
+        )
+        location = Location(latitude=lat, longitude=lon)
+        station = await hass.async_add_executor_job(
+            publi_bike.findNearestStationTo, location
+        )
+
+    coordinator = PubliBikeDataUpdateCoordinator(
+        hass, station, entry.data[BATTERY_LIMIT]
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        "coordinator": coordinator,
+    }
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
        "coordinator": coordinator,
    }
```

And will you include more items in a follow up PR? Otherwise you don't need to make it a dict, with just the coordinator. Just add the coordinator directly.

Reply:
Nothing more would be included - refactored accordingly.